pkg/dashboard/apiserver/event/event.go

```diff
@@ -97,6 +106,116 @@ func (s *Service) list(c *gin.Context) {
 	c.JSON(http.StatusOK, events)
 }
 
+// @Summary cascadeFetchEventsForWorkflow list all events for Workflow and related WorkflowNode.
+// @Description list all events for Workflow and related WorkflowNode.
+// @Tags events
+// @Produce json
+// @Param namespace query string false "The namespace of the object"
+// @Param object_id query string false "The UID of the Workflow"
+// @Param limit query string false "The max length of events list"
+// @Success 200 {array} core.Event
+// @Failure 500 {object} utils.APIError
+// @Router /events/workflow [get]
+func (s *Service) cascadeFetchEventsForWorkflow(c *gin.Context) {
+	ctx := c.Request.Context()
+	ns := c.Query("namespace")
+	objectId := c.Query("object_id")
```

Comment:
It is recommended to use param, like `/event/:id`.

Reply:
updated in ddb7075c, PTAL